### PR TITLE
Language files en_US -> de_DE (FINISHED), minor language file edits

### PR DESCRIPTION
**Pull request by keraldi**
en_Us.lang 
* had some lines that read "Undocumented", added lines for those 
* added a line for Water Ritual (that might or might not be implemented similar to Lava ritual for Raw Will).
* added documentation for the Ritual of Expulsion
* fixed some typos
* added a typo at the Water Ritual, fixed it again
* clarified Mobs -> Entities for the Ritual of Regeneration
* fixed some wrong names or explainations due to typos/wrong names and changed mechanics

de_DE.lang
* translated the whole en_US.lang including the changes I made to it.

### CODE REVIEW

**Comment by Iorce** [comment]
Did some work on de_DE and en_US. Comment for push.

**Comment by keraldi** [comment]
For "The Demon Kin" is it as in "The family of demons" (Demonkind => Dämonenartige) or "The children of demons" (Kinder der Dämonen (I would translate it as "Lilith's children" => "Liliths Kinder"))?

For "Well of Suffering" is it as in "The source of suffering" or "Fountain (A well in a village etc.) of suffering"?

For "Feathered Knife" is it as in "A knife with feathers attached to it"/"with a texture/structure that is similar to that of a feather" or "resilient, springy knife"?

**Comment by WayofTime** [comment]
"Demon Kin" means the child of demons.

Well of Suffering is closer to the first one, though it is harder to translate the full meaning of sayings/phrases.

Feathered Knife is mainly describing the fact that the knife does not kill you, just damages you. It's not a knife with feathers attached to it, so I suppose the last description you have is the most accurate.

**Comment by keraldi** [comment]
Are the teachers formal or informal with their students?
(In German, formal speech uses 2nd person plural, while informal speech uses 2nd person singular, they're the same in English (see: I, you, he/she/it, we, you, they)).

EDIT: Also, I personally don't like the sound of "Haupt-Ritualstein" (direct translation would be main ritual stone), I carried it over to my translations. Using "Meisterritualstein" would keep the abbreviation the same ("MRS" for both English and German) and it rolls off the tongue a bit better, too.

This should or would carry over to the Routing Nodes as well.

So should I change it to "Meisterritualstein" or keep "Haupt-Ritualstein"?

EDIT2: One occurence of Soul Network in the old language files translated it with "LP-Netzwerk", however I don't like putting that into the Guidebook translation. Should I use "Blood Network" ("Blutnetzwerk"), or "Soul Network" ("Seelennetzwerk")?

EDIT3: While I translated the bloodmagic/.../de_DE.lang manually, I've since switched to a machine learning translator (http://deepl.com/) for the guidebook translation (because it would take too many man hours otherwise) and I'm only reviewing that, errorchecking and making minor adjustments. There might be some differences in style/variation, which I'll try to keep constant and if I'm unsure, I'll reply with the possible translations and exact meanings.

**Comment by keraldi** [comment]
Is the tinkerer in the lava ritual story a man or a woman?
This<sup>is<sup>very<sub>important!</sub></sup></sup>

EDIT: I'm obviously not paying attention. It's a male.

**Comment by keraldi** [comment]
Entry: guide.bloodmagic.entry.ritual.wellOfSuffering.info in bloodmagicguide/.../en_US.lang:

"Damages all living non-player entities" Does it damage Zombies, Skeletons etc.? 
They're undead. I suppose the entry is incorrect. Well, they are living as in existing but are technically not living due to being undead. Not sure how I feel about it, might review it and change it a couple times.

**Comment by keraldi** [comment]
Cannot find an entry for Soul Fray in the bloodmagic/.../en_US.lang (therefore it isn't in any). This means the Soul Fray debuff won't have a translation (even in bloodmagicguide/.../xx_XX.lang)

**Comment by keraldi** [comment]
How should breaking of the fourth wall in the guidebook be handled (i.e.: "... the ritual makes no assumptions about the thing it is harvesting - it can only harvest mod-added crops if the mod in question registers a harvesting handler with the mod.") ?

I would feel personally attacked trying to translate "... mod in question registers a harvesting handler with the mod." - It feels like it doesn't belong there and nobody who is able to code would use translated terminology.

**Comment by TehNut** [comment]
Yeah I don't really think that wording should be there

**Comment by keraldi** [comment]
I would rewrite the beginning of the entry for the Ritual of Expulsion to something like this:

"As Blood Magic is a delicate art, you might find yourself in a situation in which you want to prevent the intrusion of other individuals/players/the common folk. Though worldly means exist in the form of simple traps and highly technological advancements like force fields, these can still be circumvented by some of the more crafty kind. The Ritual of Expulsion offers a more final solution that should give you the peace you need to conduct your experiments through the use of teleportation magic! ... "

In order to get rid of the fourth wall. If it is undesirable (at the very least it should not mention server....).

**Comment by keraldi** [comment]
That's all for now (Unless someone decides the guide needs more content).

File: src/main/resources/assets/bloodmagicguide/lang/de_DE.lang [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+guide.bloodmagic.welcome=Blood Magic
+
+# Page Information
+guide.bloodmagic.page.bloodAltar=Blutaltar Rezept
+guide.bloodmagic.page.soulForge=Höllenfeuerschmiede Rezept
+guide.bloodmagic.page.tier=Ausbaustufe: %d
+guide.bloodmagic.page.lp=LP: %d
+guide.bloodmagic.page.minimumWill=Minimaler Wille: %f
+guide.bloodmagic.page.drainedWill=Verbrauchter Wille: %f
+guide.bloodmagic.shapelessOrb=Formloses Orb-Rezept
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagicguide/lang/de_DE.lang]: Blutkugel here instead of Orb

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 2]
```diff
@@ -70,74 +70,150 @@ item.bloodmagic.baseComponent.reagentWater.name=Wasserreagenz
...
+item.bloodmagic.baseComponent.saltpeter.name=Saltpeter
+item.bloodmagic.baseComponent.neuro_toxin.name=Nervengift
+item.bloodmagic.baseComponent.antiseptic.name=Antiseptikum
+item.bloodmagic.baseComponent.catalyst_length_1.name=Einfacher Verlängerungskatalysator
+item.bloodmagic.baseComponent.catalyst_power_1.name=Einfacher Machtkatalysator
+
+item.bloodmagic.cutting_fluid.basic.name=Einfache Schneidelösung
+item.bloodmagic.cutting_fluid.explosive.name=Explosives Pulver
+
+item.bloodmagic.demonCrystal.default.name=Kristall dämonischen Willens
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: "Willens" is a bad translation. Think something more like willpower and volition

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 3]
```diff
@@ -70,74 +70,150 @@ item.bloodmagic.baseComponent.reagentWater.name=Wasserreagenz
...
+item.bloodmagic.demonCrystal.corrosive.name=Kristall ätzenden dämonischen Willens
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: I find "korrosiv" better than "ätzenden"

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 4]
```diff
@@ -70,74 +70,150 @@ item.bloodmagic.baseComponent.reagentWater.name=Wasserreagenz
...
-item.bloodmagic.soulSnare.base.name=Rudimentäre Falle
-item.bloodmagic.sentientBow.name=Intelligenter Bogen
+item.bloodmagic.soulSnare.base.name=Rudimentäre Schlinge
+item.bloodmagic.sentientBow.name=Empfindsamer Bogen
+item.bloodmagic.sentientArmourGem.name=Juwel der Empfindsamen Rüstung
+item.bloodmagic.sentientAxe.name=Empfindsame Axt
+item.bloodmagic.sentientPickaxe.name=Empfindsame Spitzhacke
+item.bloodmagic.sentientShovel.name=Empfindsame Schaufel
+
+item.bloodmagic.nodeRouter.name=Knoten-Router
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: "Knoten" is a bad translation for node. Use "Netzknoten" instead.

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 5]
```diff
@@ -70,74 +70,150 @@ item.bloodmagic.baseComponent.reagentWater.name=Wasserreagenz
...
+item.bloodmagic.itemFilter.exact.name=Präziser Gegenstandsfilter
+item.bloodmagic.itemFilter.ignoreNBT.name=Ignoriere NBT Gegenstandsfilter
+item.bloodmagic.itemFilter.modItems.name=Mod Gegenstandsfilter
+item.bloodmagic.itemFilter.oreDict.name=Ore Dictionary Gegenstandsfilter
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: See if there's a German translation of Ore Dictionary somewhere

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 6]
```diff
@@ -222,56 +455,295 @@ tooltip.bloodmagic.diviner.duskRune=Runen der Abenddämmerung: %d
...
+tooltip.bloodmagic.fluidFilter.exact=Filtert Flüssigkeitseingang/-ausgang
+
+tooltip.bloodmagic.fluid.type=Enthaltene Flüssigkeit: %s
+tooltip.bloodmagic.fluid.amount=Menge: %d / %d mB
+tooltip.bloodmagic.fluid.capacity=Kapazität: %d mB
+
+tooltip.bloodmagic.ghost.everything=Alles
+tooltip.bloodmagic.ghost.amount=Geistergegenstandsanzahl: %d
+
+tooltip.bloodmagic.currentType.default=Contains: Roher Wille
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Translate the "Contains:"

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang [hunk 7]
```diff
@@ -286,25 +758,141 @@ chat.bloodmagic.ritual.prevent=Das Ritual versucht, dir zu widerstehen.
...
+jei.bloodmagic.recipe.armourDowngrade=Sühne (Ritual)
 jei.bloodmagic.recipe.requiredLP=LP: %d
 jei.bloodmagic.recipe.requiredTier=Ausbaustufe: %d
-jei.bloodmagic.recipe.minimumSouls=Minimum: %d Wille
-jei.bloodmagic.recipe.soulsDrained=Verbraucht: %d Wille
+jei.bloodmagic.recipe.consumptionRate=Verbrauch: %d LP/t
+jei.bloodmagic.recipe.drainRate=Ablauf: %d LP/t
+jei.bloodmagic.recipe.minimumSouls=Minimum: %1$..2f Wille
+jei.bloodmagic.recipe.soulsDrained=Verbraucht: %1$..2f Wille
+jei.bloodmagic.recipe.lpDrained=Drained: %.d LP
```
Comment (Arcaratus) [inline src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Same here

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 2 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Do you just want to change it for the crystals, everything base (i.e. Crystal of demonic Willpower, Raw/Corrosive/Destructive/Vengeful/Steadfast Crystal of demonic Willpower, Demonic Willpower, Raw/Corrosive/Destructive/Vengeful/Steadfast demonic Willpower)?

Or every single entry of "demonic Will" in any form?

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 3 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Changed it!

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 2 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: "Willens" is a special grammar case of "Wille" both refer to the will or the "I want this" of a Person. "Willenskraft" would be the proper translation for "Willpower", however nobody would use it in such an extensive context. It would be too long and too repetitive.

In any case, I want further confirmation before changing ALL entries (because it's a lot).

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 4 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: In CS, we translate "node" in graphs as "Knoten" or "Ecke" (the latter meaning corner which I sure as hell won't use), "Netzknoten" would be a possible (and proper) translation, but again, nobody would actually refer to it with the "long" name and just refer it with a shortened version in context instead.

It is also implied that it is a network because you use those things in a network that transfers stuff. I'll change it for the tool (Node-Router) but I want further confirmation before changing it for the other stuff.

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 5 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Since the Ore Dictionary is a Forge thing and you'd rarely see someone insane enough to translate stuff that is used in programming itself, I doubt that I'll find one. An initial search didn't yield any results.

And the translation is "Erzwörterbuch" which sounds stupid.

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 6 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Done!

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 7 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Done!

File: src/main/resources/assets/bloodmagic/lang/de_DE.lang (hunk 2 above)
Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: from google translate: 

  | der Wille | will, intention

And the quality of that "intention" is the power it has.
Or so I think.

Comment (Arcaratus) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Although it does have that meaning, it's a bit obscure. I think it's just easier to translate it to "Will" as a borrowed noun.

Comment (keraldi) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Leaving it untranslated would make it even more obscure because people won't expect an untranslated word there, especially if it has meaning in German.

Translating it as "Willenskraft" /"Will power" for basic things (The will drops/crystals) would make the meaning clear while preventing the exhaustive spam of long words and repetitions if every single entry would be translated.

Comment (Arcaratus) [reply src/main/resources/assets/bloodmagic/lang/de_DE.lang]: Sounds good.

**Comment by keraldi** [comment]
Updated to incorporate Arcaratus' quick fixes. Please review and merge :3

File: gradle.properties [hunk 8]
```diff
@@ -4,9 +4,7 @@ mod_version=2.2.12
 mc_version=1.12.2
 forge_version=14.23.2.2611
 curse_id=224791
-
```
Comment (TehNut) [inline gradle.properties]: Remove the changes to this file.

File: src/main/java/WayofTime/bloodmagic/livingArmour/downgrade/LivingArmourUpgradeDisoriented.java [hunk 9]
```diff
@@ -9,7 +9,7 @@
 
 public class LivingArmourUpgradeDisoriented extends LivingArmourUpgrade {
     public static final int[] costs = new int[]{-10, -20, -30, -40, -70, -80, -100, -140, -180, -220};
-    public static final double[] chance = new double[]{0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.008, 0.0010, 0.0012, 0.014};
+    public static final double[] chance = new double[]{0.001, 0.002, 0.003, 0.004, 0.005, 0.006, 0.008, 0.010, 0.012, 0.014};
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/livingArmour/downgrade/LivingArmourUpgradeDisoriented.java]: This change should be moved to it's own PR

**Review by TehNut — CHANGES_REQUESTED** [review]
